src/core/linter-rules/check-period-in-p.js

```diff
@@ -0,0 +1,55 @@
+/**
+ * Linter rule "check-period". Makes sure the there are no periods missing at the end of a <p>
+ *   in the ReSpec config.
+ */
+import { lang as defaultLang } from "core/l10n";
+import LinterRule from "core/LinterRule";
+
+const name = "check-period-in-p";
+
+const meta = {
+  en: {
+    description: "`<p>` tags should end with a period",
+    howToFix: "Please put a period at the end of this `<p>` tag",
+  },
+};
+// Fall back to english, if language is missing
+const lang = defaultLang in meta ? defaultLang : "en";
+
+/**
+ * Runs linter rule.
+ *
+ * @param {Object} config The ReSpec config.
+ * @param  {Document} doc The document to be checked.
+ */
+function lintingFunction(conf, doc) {
+  console.log("dsfdf")
+  
+  const allElements = Array.from(doc.querySelectorAll("p"));
+  console.log(allElements);
+  const offendingElements = Array.from(doc.querySelectorAll("p"))
+    // this gives us all paragraphs that don't end with a period
+    .filter(elem => !elem.textContent.trim().endsWith("."))
```

Comment:
note that we are doing same operations over and over again here, which is not ideal (filter, access textContent, trim, check ending). 

We can combine the the punctuation checks with a regex like `[.!?:]\s*$` (ends with any these characters, and zero or more spaces).

Reply:
The regex needs to work multiline too.

Reply:
Ohk. No problem. I'll try the regex :+1:

Reply:
@marcoscaceres I tried using the regex. I had never used it before so it took some time to understand. I guess its working now. Please have a look. :smile:

Reply:
Ah yeah, regexes are the worst thing ever. But can be useful in situations like this. I live in:

https://www.regexpal.com

There are other useful sites that help unpack/write regexes.